File: SourceCode/JavaScript/Demo/UpDown.js
"use strict";
/*global console */
// run this using node, as in: node UpDown.js
var runs = 1000000;

function updown() {
  var i, j;
  var count = 1000;
  console.log("Doing " + count + " times");

  var time1 = new Date();
  var sum1 = 0;
  for (i = 0; i < runs; i++) {
    for (j = 0; j < count; j++)
      { sum1++; }
  }
  var time2 =  new Date();
  var sum2 = 0;
  for (i = runs; i > 0; i--) {
    for (j = count; j > 0; j--)
      { sum2++; }
  }
  var time3 = new Date();
  var sum3 = 0;
  for (i = runs; i !== 0; i--) {
    for (j = count; j !== 0; j--)
      { sum3++; }
  }
  var time4 = new Date();

  var sum4 = 0; i = 0;
  while (i !== runs) {
    j = 0;
    while (j !== count)
      { sum4++; j++; }
    i++;
  }
  var time5 = new Date();

  var sum5 = 0; i = 0;
  while (i < runs) {
    j = 0;
    while (j < count)
      { sum5++; j++; }
    i++;
  }
  var time6 = new Date();

  console.log("Up time is " + (time2 - time1) +
          ", down time is " + (time3 - time2));
  console.log("Check vs. zero time is " + (time4-time3));
  console.log("While loop time is " + (time5-time4));
  console.log("While loop 2 time is " + (time6-time5));
  console.log("Sum1 is " + sum1 + ", sum2 is " + sum2 + ", sum 3 is " + sum3);
  console.log("Sum4 is " + sum4);
  console.log("Sum5 is " + sum5);
}

updown();

File: SourceCode/Python/Demo/UpDown.py
runs = 1000

count = 1000
print "Doing ", count, " times"

time1 = datetime.now()
sum1 = 0
for i in xrange(0, runs):
  for j in xrange(0, count):
    sum1 = sum1 + 1
time2 = datetime.now()

sum2 = 0;
for i in reversed(xrange(0, runs)):
  for j in reversed(xrange(0, count)):
    sum2 = sum2 + 1
time3 = datetime.now()

sum3 = 0; i = 0; j = 0;
while i != runs:
  j = 0;
  while j != count:
    sum3 = sum3 + 1;
    j = j + 1;
  i = i + 1;
time4 = datetime.now()

sum4 = 0; i = 0; j = 0;
while i < runs:
  j = 0;
  while j < count:
    sum4 = sum4 + 1;
    j = j + 1;
  i = i + 1;
time5 = datetime.now()

print "Up time is ", millis(time2 - time1)
print "Down time is ", millis(time3 - time2)
print "While time is ", millis(time4 - time3)
print "While time 2 is ", millis(time5 - time4)

print "Sum1 is ", sum1, ", sum2 is ", sum2
print "Sum3 is ", sum3
print "Sum4 is ", sum4
